### PR TITLE
PT-2195: Some JSON is exported even if nothing is marked for export 

### PR DESCRIPTION
**Pull request by antonkatz**
@allasm @itaiGershtansky  please review

### CODE REVIEW

File: components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/ClinicalStatusController.java [hunk 1]
```diff
@@ -90,6 +91,10 @@ public String getName()
     @Override
     public void writeJSON(Patient patient, JSONObject json, Collection<String> selectedFieldNames)
     {
+        if (selectedFieldNames != null && !selectedFieldNames.contains(UNAFFECTED)) {
```
Comment (antonkatz) [inline components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/ClinicalStatusController.java]: Ideally, this should check if ```selectedFieldNames``` contains ```getName()```, but the export UI doesn't work that way.

File: components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/ClinicalStatusController.java [hunk 2]
```diff
@@ -66,7 +68,6 @@ public String getName()
     @Override
     public PatientData<String> load(Patient patient)
     {
-        String unaffected = "unaffected";
         String affected = "affected";
```
Comment (buske) [inline components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/ClinicalStatusController.java]: I would move affected to a static variable as well for consistency.

File: components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/ClinicalStatusController.java (hunk 1 above)
Comment (buske) [inline components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/ClinicalStatusController.java]: Isn't the field called `clinicalStatus`?

File: components/patient-data/impl/src/test/java/org/phenotips/data/internal/controller/ClinicalStatusControllerTest.java [hunk 3]
```diff
@@ -168,8 +169,7 @@ public void writeJSONWithSelectedFieldsAddsAffectedValueToJSON() throws Componen
             new SimpleValuePatientData<>(DATA_NAME, AFFECTED);
         doReturn(data).when(this.patient).getData(DATA_NAME);
         JSONObject json = new JSONObject();
-        Collection<String> selectedFields = new LinkedList<>();
-        selectedFields.add(DATA_NAME);
+        Collection<String> selectedFields = Arrays.asList(UNAFFECTED);
```
Comment (buske) [inline components/patient-data/impl/src/test/java/org/phenotips/data/internal/controller/ClinicalStatusControllerTest.java]: Why is this UNAFFECTED and not DATA_NAME?

File: components/patient-data/impl/src/test/java/org/phenotips/data/internal/controller/ClinicalStatusControllerTest.java [hunk 4]
```diff
@@ -196,8 +210,7 @@ public void writeJSONWithSelectedFieldsAddsUnaffectedValueToJSON() throws Compon
         PatientData<String> data = new SimpleValuePatientData<>(DATA_NAME, UNAFFECTED);
         doReturn(data).when(this.patient).getData(DATA_NAME);
         JSONObject json = new JSONObject();
-        Collection<String> selectedFields = new LinkedList<>();
-        selectedFields.add(DATA_NAME);
+        Collection<String> selectedFields = Arrays.asList(UNAFFECTED);
```
Comment (buske) [inline components/patient-data/impl/src/test/java/org/phenotips/data/internal/controller/ClinicalStatusControllerTest.java]: Again, why UNAFFECTED?

**Comment by buske** [comment]
I think this confounds the distinction between `unaffected` the status and `unaffected` the field name. I'd suggest having separate variables for each (even if the have the same value) to make this distinction very clear. Both here, and in the tests.

**Comment by allasm** [comment]
Looks good other than +1 to @buske comment above

**Comment by antonkatz** [comment]
@buske's comment was addressed in the code. Not sure why the last commit is showing up before his comment.

**Comment by allasm** [comment]
I meant "I would move "affected" to a static variable as well for consistency", it doesn't seem to have been done?

**Comment by antonkatz** [comment]
Done

**Comment by allasm** [comment]
I think this pull request has been merged and can be deleted?

**Pull request closed without merging**